pinot-core/src/test/java/org/apache/pinot/core/operator/ThreadCpuTimeMeasurementTest.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.core.operator;
+
+import org.locationtech.jts.util.Assert;
+import org.testng.annotations.Test;
+
+
+public class ThreadCpuTimeMeasurementTest {
+
+  @Test
+  public void testAdditionTransformFunction() {
+    class testCase {
+      final long totalWallClockTimeNs;
+      final long multipleThreadCpuTimeNs;
+      final int numServerThreads;
+      final long totalThreadCpuTimeNs;
+
+      testCase(long totalWallClockTimeNs, long multipleThreadCpuTimeNs, int numServerThreads,
+          long totalThreadCpuTimeNs) {
+        this.totalWallClockTimeNs = totalWallClockTimeNs;
+        this.multipleThreadCpuTimeNs = multipleThreadCpuTimeNs;
+        this.numServerThreads = numServerThreads;
+        this.totalThreadCpuTimeNs = totalThreadCpuTimeNs;
+      }
+    }
+
+    testCase[] testCases =
+        new testCase[]{
+            new testCase(4245673, 7124487, 3, 8995331),
```

Comment:
Can you please take a look at the production data we collected during verification and then try to make sure everything is covered ?